### PR TITLE
feat(kit): `InputYear` uses `Maskito` instead of legacy `text-mask`

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?
Partially solves #120

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/kit/components/input-year/input-year.component.ts [hunk 1]
```diff
@@ -88,8 +84,20 @@ export class TuiInputYearComponent
         return this.options.icon;
     }
 
+    @tuiPure
+    getMaskOptions(max: number): MaskitoOptions {
+        const numberOptions = maskitoNumberOptionsGenerator({
```
Comment (waterplea) [inline projects/kit/components/input-year/input-year.component.ts:89]: Looks like unnecessary const to me :)

File: projects/kit/components/input-year/input-year.component.ts [hunk 2]
```diff
@@ -88,8 +84,20 @@ export class TuiInputYearComponent
...
+            max,
+            precision: 0,
+            isNegativeAllowed: false,
```
Comment (waterplea) [inline projects/kit/components/input-year/input-year.component.ts:92]: Do `precision` and `isNegativeAllowed` matter if we override the `mask` anyway?

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/input-year/input-year.component.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/kit/components/input-year/input-year.component.ts:92]: You are right, I should drop it.

Comment (nsbarsukov) [reply projects/kit/components/input-year/input-year.component.ts]: fixed

File: projects/kit/components/input-year/input-year.component.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/kit/components/input-year/input-year.component.ts]: fixed

**Review by splincode — APPROVED** [review]

**Pull request merged**